src/mbgl/style/layers/symbol_layer.cpp

```diff
@@ -169,7 +169,7 @@ void SymbolLayer::setIconSize(PropertyValue<float> value) {
     if (value == getIconSize())
         return;
     impl->layout.iconSize.set(value);
-    impl->observer->onLayerLayoutPropertyChanged(*this);
+    impl->observer->onLayerLayoutPropertyChanged(*this, true);
```

Comment:
`symbol_layer.cpp` is a generated file, so this change will get overwritten.

Instead of a boolean, how about a `const char * property`, with name from style spec, e.g. `"icon-size"`. We can update `layer.cpp.ejs` to add this parameter for every setter, and do exact matches in `Style::onLayerLayoutPropertyChanged`.